packages/astro/src/core/render/dev/vite.ts

```diff
@@ -82,7 +82,7 @@ export async function* crawlGraph(
 						}
 					}
 				}
-				if (!isPropagationStoppingPoint) {
+				if (isImportedBy(id, importedModule) && !isPropagationStoppingPoint) {
 					importedModules.add(importedModule);
 				}
 			}
```

Comment:
On line 47, maybe we can do this instead?

```diff
-			for (const importedModule of entry.importedModules) {
+			for (const importedModule of entry.ssrTransformResult.deps) {
```

Assuming `ssrTransformResult` would always exist since we're running the Astro component in SSR. This is a bit similar to https://github.com/withastro/astro/pull/6716 when I was refactoring it.

Tailwind strikes again 😦

Reply:
I don't think this is a problem with Tailwind so much. They aren't doing anything wrong here. This comes down to the Vite API changing and us using the new one incorrectly. 

Thanks @bluwy I'll give that option a try.

Reply:
`ssrTransformResult.deps` will work. It is urls (paths), so need to compare against `entry.url`.